conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1129,3 +1129,100 @@ def test_cmake_toolchain_vars_when_option_declared():
     t.run_command("cmake -S . -B build/ -DBUILD_SHARED_LIBS=ON")
     assert "mylib target type: SHARED_LIBRARY" in t.out
     assert "mylib position independent code: ON" in t.out
+
+
+#@pytest.mark.tool_cmake()
+def test_find_program_for_tool_requires():
+    """Test that the same reference can be both a tool_requires and a regular requires,
+    and that find_program (executables) and find_package (libraries) find the correct ones
+    when cross building.
+    """
+
+    client = TestClient()
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        class TestConan(ConanFile):
+            name = "foobar"
+            version = "1.0"
+            settings = "os", "arch", "compiler", "build_type"
+            exports_sources = "*"
+            def layout(self):
+                pass
+            def package(self):
+                self.copy(pattern="lib*", dst="lib")
+                self.copy(pattern="*bin", dst="bin")
+    """)
+
+    host_profile = textwrap.dedent("""
+    [settings]
+        os=Linux
```

Comment:
If this is a functional test, this will probably be broken in Win/OSX? need a ``pytest.skip``?

Reply:
This doesn't really test functionality that is platform specific - it's "cmake only" - figured that by fixing the profiles and setting `CMAKE_CXX_COMPILER_WORKS` in CMake (I've seen this done in other tests, this would enable us to runt this test on all platforms, or at least to write it in a way that is more platform agnostic

Reply:
I thought that providing a profile like that for Windows would fail at some point, like the armv8 architecture, or something like that. Probably can be simplified and left mostly with the platform specific ones for testing, but I am ok with it as-is

Reply:
oh, the host profile could be listing a random architecture here that is not x86_64 - this test is not exercising anything that is platform specific, hence I feel it's safe to "fix" the profiles like many other tests do. A different approach could be tried if we want the profiles to match something that makes more sense on the machine the test is running on.